src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSFilter.java

```diff
@@ -0,0 +1,334 @@
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.filters.AmbiguousBaseReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadLengthReadFilter;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.SVUtils;
+import org.broadinstitute.hellbender.tools.spark.utils.ReadFilterSparkifier;
+import org.broadinstitute.hellbender.tools.spark.utils.ReadTransformerSparkifier;
+import org.broadinstitute.hellbender.transformers.BaseQualityClipReadTransformer;
+import org.broadinstitute.hellbender.transformers.BaseQualityReadTransformer;
+import org.broadinstitute.hellbender.transformers.DUSTReadTransformer;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemIndexSingleton;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.SAMRecordToGATKReadAdapter;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.*;
+
+/**
+ * Performs PathSeq filtering steps and manages associated resources.
+ */
+public final class PSFilter implements AutoCloseable {
+
+    private final JavaSparkContext ctx;
+    private final PSFilterArgumentCollection filterArgs;
+    private final MetricsState metricsState;
+    private JavaRDD<GATKRead> reads;
+    private final SAMFileHeader header;
+
+    public PSFilter(final JavaSparkContext ctx, final PSFilterArgumentCollection filterArgs, final JavaRDD<GATKRead> inputReads,
+                    final SAMFileHeader header) {
+        Utils.nonNull(ctx);
+        Utils.nonNull(filterArgs);
+        Utils.nonNull(inputReads);
+        this.ctx = ctx;
+        this.filterArgs = filterArgs;
+        this.metricsState = initializeMetics(filterArgs.metricsFileUri);
+        this.reads = inputReads;
+        this.header = header;
+    }
+
+    @VisibleForTesting
+    static JavaRDD<GATKRead> setPairFlags(final JavaRDD<GATKRead> reads, final int readsPerPartitionGuess) {
+        return repartitionPairedReads(reads).mapPartitions(iter -> setPartitionUnpairedFlags(iter, readsPerPartitionGuess));
+    }
+
+    private static JavaRDD<GATKRead> clearAllAlignments(final JavaRDD<GATKRead> reads, final SAMFileHeader header) {
+        return reads.map(read -> clearReadAlignment(read, header));
+    }
+
+    /**
+     * Returns input read with alignment-related info cleared
+     */
+    private static GATKRead clearReadAlignment(final GATKRead read, final SAMFileHeader header) {
+        final GATKRead newRead = new SAMRecordToGATKReadAdapter(new SAMRecord(header));
+        newRead.setName(read.getName());
+        newRead.setBases(read.getBases());
```

Comment:
This could change things with downstream bwa alignment. Thanks for pointing this out - I've added a check now that sets the read back to its original orientation.